### DISCUSSION TITLE
fatal error: CompatibilityTable.h: No such file or directory

### DISCUSSION THREAD

**Discussion by grey1808** [ZZ-Cat/CRSFforArduino#19] [General]
Greetings my friend! Please tell me what am I doing wrong. I'm sorry, I'm new to arduino and maybe the question is very obvious, but I can't figure it out.

I included the library and decided to compile your example, but it didn't work because I get an error at the compilation stage:

`Arduino: 1.8.19 (Linux), Pay:"Arduino M0"
/home/grey/Arduino/libraries/CRSFforArduino-Main-Trunk/src/lib/CRSFforArduino/CRSFforArduino.cpp:28:10: fatal error: CompatibilityTable.h: No such file or directory
 #include "CompatibilityTable.h"
          ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
exit status 1
`
Please tell me how to connect this CompatibilityTable.h

**Comment by ZZ-Cat** [comment]
That's weird. 🤔

I tried to build `channels.ino` myself in the Arduino IDE (version 1.8.19), & it showed the same compiler error.
It builds perfectly fine in PlatformIO. It might take me a little bit to figure out why the Arduino IDE is being finicky with it.

I created #20 from this, so I can investigate it. I know for a fact that the Arduino IDE _hates_ having nested folders & files in a given library's folder directory & favours having _all_ of the headers & c++ source files all in one `src` folder (which is disorganised, & it irritates me like a bad rash). That's one of the reasons why I ported over to PlatformIO, yet I tried to maintain backward compatibility with the old Arduino IDE.

**Comment by ZZ-Cat** [comment]
Okay, I have found the problem. The good news is that it's not what you're doing on your end @grey1808, thankfully. You encountered a compilation bug in my library (which I am fixing, right now, as I write this reply).
As I have previously stated, the Arduino IDE _hates_ libraries with subfolders, so I have had to create a clunky workaround for the Arduino IDE to compile _with_ the compatibility table included.

I have created #21 for you to try out & see if that works for you. You can check it out [here](https://github.com/ZZ-Cat/CRSFforArduino/tree/ZZ-Cat/issue20)

**Reply by grey1808** [reply]
Thank you my friend! It seems to be working! I'm not very good at arduino, but it seems I found an error in the example, after "CRSFforArduino crsf = CRSFforArduino(&Serial1)" there is no ";" , but this is a trifle. It seems that your corrected [library](https://github.com/ZZ-Cat/CRSFforArduino/tree/ZZ-Cat/issue20)  works, but if anything, I will deal with PlatformIO. Thank you! I didn’t expect you to answer) I thought I wouldn’t succeed, I’ll try. If I find something else, I'll write to you)

By the way, after compilation, the compiler displays messages:
`/home/grey/.arduino15/packages/arduino/tools/arm-none-eabi-gcc/7-2017q4/bin/../lib/gcc/arm-none-eabi/7.2.1/../.. /../../arm-none-eabi/bin/ld: warning: changing start of section .bss by 4 bytes
/home/grey/.arduino15/packages/arduino/tools/arm-none-eabi-gcc/7-2017q4/bin/../lib/gcc/arm-none-eabi/7.2.1/../.. /../../arm-none-eabi/bin/ld: warning: changing start of section .bss by 4 bytes
/home/grey/.arduino15/packages/arduino/tools/arm-none-eabi-gcc/7-2017q4/bin/../lib/gcc/arm-none-eabi/7.2.1/../.. /../../arm-none-eabi/bin/ld: warning: changing start of section .bss by 4 bytes
/home/grey/.arduino15/packages/arduino/tools/arm-none-eabi-gcc/7-2017q4/bin/../lib/gcc/arm-none-eabi/7.2.1/../.. /../../arm-none-eabi/bin/ld: warning: changing start of section .bss by 4 bytes
/home/grey/.arduino15/packages/arduino/tools/arm-none-eabi-gcc/7-2017q4/bin/../lib/gcc/arm-none-eabi/7.2.1/../.. /../../arm-none-eabi/bin/ld: warning: changing start of section .bss by 4 bytes`

It looks like it's just a warning and it doesn't mean anything.

**Reply by ZZ-Cat** [reply]
I merged #21 into the Main-Trunk branch, & closed #20 because it's completed.
While I was at it, I fixed the typo that you pointed out in the example that I provided in the readme.

Your compiler warnings there are outside of the scope of CRSF for Arduino. IIRC, it's how GCC manages the flash memory of our modern SAM D21, D51, & E51 microcontrollers through smart wear levelling. Those warnings are benign. I have encountered them myself, after [x] amount of times I have built firmware for any of my projects. It's safe to ignore them.

Thanks for pointing that out, by the way.
If you encounter any more bugs, or if you have any suggestions for CRSF for Arduino for next time... please open up a [ticket](https://github.com/ZZ-Cat/CRSFforArduino/issues/new/choose) in my [Issues](https://github.com/ZZ-Cat/CRSFforArduino/issues) tab.
The Discussions area is primarily for general feedback that falls outside of the scope of Issues & Pull Requests.

I do accept feedback, & I do my best to respond in a timely manner (sometimes not straight away, but as soon as I can). =^/,..,^=
